### DISCUSSION TITLE
Missing wheels for M1 (Mac ARM) architecture

### DISCUSSION THREAD

**Discussion by thomasaarholt** [Toblerity/Fiona#1071] [General]
## Expected behavior and actual behavior.

I expected to be able to install fiona with pip in python 3.9 on an M1 (arm architecture) Mac.

## Steps to reproduce the problem.

`pip install fiona` raises the following error:

```bash
❯ pip install fiona
Collecting fiona
  Using cached Fiona-1.8.21.tar.gz (1.0 MB)
    ERROR: Command errored out with exit status 1:
     command: /Users/thomas/.asdf/installs/python/3.9.9/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/yt/z22_f0896l74f8gs3_20vc_00000gn/T/pip-install-456dvi2q/fiona_ecb5449ed2884c3db55434cbb048b28b/setup.py'"'"'; __file__='"'"'/private/var/folders/yt/z22_f0896l74f8gs3_20vc_00000gn/T/pip-install-456dvi2q/fiona_ecb5449ed2884c3db55434cbb048b28b/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/yt/z22_f0896l74f8gs3_20vc_00000gn/T/pip-pip-egg-info-tq10nup4
         cwd: /private/var/folders/yt/z22_f0896l74f8gs3_20vc_00000gn/T/pip-install-456dvi2q/fiona_ecb5449ed2884c3db55434cbb048b28b/
    Complete output (2 lines):
    Failed to get options via gdal-config: [Errno 2] No such file or directory: 'gdal-config'
    A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/67/5c/4e028e84a1f0cb3f8a994217cf2366360ca984dfc1433f6171de527d0dca/Fiona-1.8.21.tar.gz#sha256=3a0edca2a7a070db405d71187214a43d2333a57b4097544a3fcc282066a58bfc (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
## Operating system

Mac OS X 12.2.1.

This is in the same vein as #1028. It would be great to add arm architecture support so that fiona and its 247 downstream packages can be used on arm :) 

The linux arm package is of interest for mac as well, since docker images that run on the M1 architecture will run arm arch linux. (So if you run a docker image natively on M1 mac, the current linux package will not be installed)

**Comment by sgillies** [comment]
Fiona 1.8.21 wheels for x86_64 manylinux2014 and macos (10.10) only were built on GitHub actions using https://github.com/sgillies/fiona-wheels.

I get it @thomasaarholt, and I'm sorry to have no good news for you today.

In the M1 case, we're mainly blocked by the lack of free CI with native M1 workers. Cross-compilation is, in theory, possible, and I've done it for the GEOS library, but there are a lot of unknowns for the deep stack of libraries that GDAL depends on.

For the time being: use packages from conda-forge. The conda-forge project's mission is to build packages. You can even sponsor the project with money to do it. The Fiona project is not currently on a mission to build packages for every available platform. This is a software code project, not a packaging or distribution project.

**Reply by futurebenmorris** [reply]
I'm also running in to issues with installation on Apple Silicon (and MacOS 13 Ventura). Looks like Github have M1 runners in beta (see [here](https://github.com/github/roadmap/issues/528)). Using conda-forge is simply not an option for me at the moment sadly. The very nature of being a software project means it should run on an array of architectures. I don't think that's an unreasonable requirement. Any way round this other than "use conda-forge" in the meantime would be a great help to quite a few folks I'm sure!

**Comment by christianlehre** [comment]
@thomasaarholt I encountered the same issue. As you mentioned the installation complaints about a missing GDAL config. Try installing gdal with brew before pip-installing fiona again. Have a look at [this](https://stackoverflow.com/questions/71137617/error-installing-geopandas-in-python-on-mac-m1) thread on stack overflow :)
